### DISCUSSION TITLE
Trouble configuring Nginx with \rpc\ function

### DISCUSSION THREAD

**Discussion by Mjaffres** [PostgREST/postgrest#1853] [Q&A] [answered]
Hello,

I am trying to call a fonction through postgrest
it works on port 3001 where my api for postgrest is

but doesn't work with my nginx server
otherwise any call I do on table\views on my api schema is doing great.

the test request in JS 

```
fetch('/rpc/transformed_network_height_velocity', {
                method:'POST', 
                headers:new Headers({"Authorization": "Bearer "+token}), 
                body: JSON.stringify({transfo:transfo[tf].id, station:id})
            }).then(function(response) {
                if (response.status != 201){
                    alert(response.statusText);
                }
            });
```

the backend in SQL is no problem, it works if I test the DB with PGAdmin for exemple.

for exemple this work : 
http://localhost:3001/rpc/transformed_network_height_velocity?station=yo&transfo=69444e58-9e6e-4b24-8422-0ba11703fd57
all call with api
http://localhost:3001/api/stations
http://localhost:8282/api/stations

this doesn't  : 
call with rpc on nginx redirect
http://localhost:8282/rpc/transformed_network_height_velocity

my nginx conf looks like the exemple :
```
---------------------------------------------

upstream postgrest {
        server localhost:3001;
        keepalive 64;
    }

server {
       listen       8282;
       server_name  somename  alias  another.alias;
       client_max_body_size 100M;

       location / {
            root C:\diagperm\diag_perm;
            index  index.html index.htm;
        }

        location /api/ {
            default_type  application/json;
            proxy_hide_header Content-Location;
            add_header Content-Location  /api/$upstream_http_content_location;
            proxy_set_header  Connection "";
            proxy_http_version 1.1;
            proxy_pass http://postgrest/;
        }

    }

```

I think something is not good in this nginx conf, I should add rpc redirection ...
but I don't see how this works and what I should do here
any explanation or redirection to the good doc would be much appreciated.

Thanks

**Comment by steve-chavez** [comment, marked as answer]
Hey @Mjaffres,

I think you're missing the `api` prefix for your RPC endpoint, it should be:

```
http://localhost:8282/api/rpc/transformed_network_height_velocity
```

**Reply by Mjaffres** [reply]
Hi, 

indeed , I didn't  understood that the correct way was to put schema name then rpc in the query.

so this works like a charm : 

http://localhost:3001/api/rpc/transformed_network_height_velocity?station=yo&transfo=69444e58-9e6e-4b24-8422-0ba11703fd57

Thanks
